core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryBuilder.java

```diff
@@ -20,6 +23,8 @@
 import org.uniprot.core.uniprotkb.xdb.UniProtKBCrossReference;
 
 public class UniProtKBEntryBuilder implements Builder<UniProtKBEntry> {
+    public static final String COUNT_BY_COMMENT_TYPE_ATTRIB = "countByCommentType";
```

Comment:
Maybe use an Enum instead of the constants, in this new map field we will have more 3 or 4 fields fields, and we could use it to restrict in the builder, what do you think about this idea? Also see the comment at extraAttributesAdd method that complement the idea.. What do you think?